components/net_traits/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use regex::Regex;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use std::string::String;
+
+const BLACKLIST_ERR1: &'static str = "This UUID is not valid";
+const BLACKLIST_ERR2: &'static str = "The filling of the blacklist resulted in error";
+const EXCLUDE_READS: &'static str = "exclude-reads";
+const EXCLUDE_WRITES: &'static str = "exclude-writes";
+const VALID_UUID_REGEX: &'static str = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}";
+
+pub struct BluetoothBlacklist {
+    blacklist: HashMap<String, Value>,
+}
+
+#[derive(Eq, PartialEq)]
+pub enum Value {
+    Exclude,
+    ExcludeReads,
+    ExcludeWrites,
+}
+
+impl BluetoothBlacklist {
+    pub fn new () -> BluetoothBlacklist {
+        BluetoothBlacklist {
+            blacklist: HashMap::new(),
+        }
+    }
+
+    pub fn is_excluded(&mut self, uuid: &str) -> Result<bool, String> {
+        if self.blacklist.is_empty() {
+            try!(self.fill_blacklist());
+        }
+        if let Some(value) = self.blacklist.get(uuid) {
+            return Ok((value.eq(&Value::Exclude)));
+        }
+        Ok(false)
+    }
+
+    pub fn is_excluded_from_reads(&mut self, uuid: &str) -> Result<bool, String> {
+        if self.blacklist.is_empty() {
+            try!(self.fill_blacklist());
+        }
+        if let Some(value) = self.blacklist.get(uuid) {
+            return Ok((value.eq(&Value::ExcludeReads)));
+        }
+        Ok(false)
+    }
+
+    pub fn is_excluded_from_writes(&mut self, uuid: &str) -> Result<bool, String> {
+        if self.blacklist.is_empty() {
+            try!(self.fill_blacklist());
+        }
+        if let Some(value) = self.blacklist.get(uuid) {
+            return Ok((value.eq(&Value::ExcludeWrites)));
+        }
+        Ok(false)
+    }
+
+    fn fill_blacklist(&mut self) -> Result<(), String> {
+        let path = Path::new("./blacklist.txt");
+        let mut file = match File::open(&path) {
+            Ok(file) => file,
+            Err(_) => return Err(format!("Couldn't open {}", path.display())),
+        };
+        let mut data = String::new();
+        let _ = file.read_to_string(&mut data);
+        for line in data.lines() {
+            let linestr = line.to_string();
+            if linestr.starts_with('#') || linestr.is_empty() {
```

Comment:
we should check if it's empty first
`if linestr.is_empty() || linestr.starts_with('#') {`